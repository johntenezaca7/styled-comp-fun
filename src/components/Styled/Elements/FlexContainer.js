import styled from "styled-components";

export const FlexContainer = styled.div`
	display: flex;
	flex-direction: ${props => props.flow };
	width: ${props => props.width };
	justify-content: ${props => props.jContent };
	align-items: ${props => props.aItems }};
	margin: ${props => props.margin };
	padding: ${props => props.padding };
`;

FlexContainer.defaultProps = {
	flow: "row",
	width: "100%",
	jContent: "flex-start",
	aItems: "flex-start",
	margin: "0",
	padding: "0"
};

