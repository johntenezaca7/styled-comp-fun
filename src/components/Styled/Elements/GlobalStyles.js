import { createGlobalStyle } from "styled-components";

export const GlobalStyles = createGlobalStyle`
  body {
    @import url('https://fonts.googleapis.com/css?family=Exo&display=swap');
    padding: 0;
    margin: 0;
    font-family: 'Exo', sans-serif;

    #root  {
      margin: 0;
    }
  }
`;