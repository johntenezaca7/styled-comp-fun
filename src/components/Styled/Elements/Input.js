import styled from "styled-components";

export const Input = styled.input`
	border: ${props => props.theme.color } 1px solid;
	height: 2rem;
`;

Input.defaultProps = {
	theme: {
		color: "palevioletred"
	}
};