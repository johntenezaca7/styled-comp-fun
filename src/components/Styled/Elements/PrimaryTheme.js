export const PrimaryTheme = {
	color: "mediumseagreen"
};