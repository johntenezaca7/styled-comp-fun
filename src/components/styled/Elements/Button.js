import styled from "styled-components";

export const Button = styled.button`
  font-size: 1em;
  border-radius: 3px;
  cursor: pointer;
  color: ${props => props.theme.color};
  border: 2px solid ${props => props.theme.color};
  padding: ${props => props.padding };
  margin: ${props => props.margin };
`;

Button.defaultProps = {
  padding: "0.25rem 1rem",
  margin: "0",
  theme: {
    color: "#70A288"
  },
};

