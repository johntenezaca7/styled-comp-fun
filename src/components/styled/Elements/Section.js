import styled from "styled-components";

export const Section = styled.section`
  max-width: 99rem;
  padding: 2rem;
  background-color: #ffffff;
  height: 100vh;
`;
