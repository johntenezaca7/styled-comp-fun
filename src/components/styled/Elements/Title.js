import styled from "styled-components";

export const Title = styled.h1`
  color: ${ props => props.theme.main };
  font-size: 2.5rem;
  margin-top: 0;
  margin-bottom: 12px;
`;

// We are passing a default theme for Buttons that arent wrapped in the ThemeProvider
Title.defaultProps = {
  theme: {
    main: "palevioletred"
  }
}
