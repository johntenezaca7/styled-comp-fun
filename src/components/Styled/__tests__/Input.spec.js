import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import { Input } from "../Elements/Input";

test("Input renders correctly.", () => {
  const element = renderer.create(<Input/>).toJSON();
  expect(element).toMatchSnapshot();
});

test("Default props are passed", () => {

});
