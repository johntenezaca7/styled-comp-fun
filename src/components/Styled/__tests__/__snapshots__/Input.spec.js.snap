// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly. 1`] = `
.c0 {
  border: palevioletred 1px solid;
  height: 2rem;
}

<input
  className="c0"
/>
`;
